api/jsonrpc/server.go

```diff
@@ -175,58 +175,66 @@ func (j *JSONRPCServer) Execute(
 	defer span.End()
 
 	actionRegistry := j.vm.ActionRegistry()
-	action, err := (*actionRegistry).Unmarshal(codec.NewReader(args.Action, len(args.Action)))
-	if err != nil {
-		return fmt.Errorf("failed to unmashal action: %w", err)
+	actions := make([]chain.Action, 0)
+	for _, action := range args.Actions {
+		action, err := (*actionRegistry).Unmarshal(codec.NewReader(action, len(action)))
+		if err != nil {
+			return fmt.Errorf("failed to unmashal action: %w", err)
+		}
+		actions = append(actions, action)
 	}
 
 	now := time.Now().UnixMilli()
 
-	// Get expected state keys
-	stateKeysWithPermissions := action.StateKeys(args.Actor)
+	storage := make(map[string][]byte)
+	ts := tstate.New(1)
 
-	// flatten the map to a slice of keys
-	storageKeysToRead := make([][]byte, 0)
-	for key := range stateKeysWithPermissions {
-		storageKeysToRead = append(storageKeysToRead, []byte(key))
-	}
+	for _, action := range actions {
```

Comment:
Could we update this to exactly match the semantics in `chain/processor.go` to make sure that the behavior aligns with tx execution?

In tx execution, we gather all of the storage keys from all included actions first, create the tstate view instance, and then iterate and execute each action.

This results in a change in behavior between tx execution and simulation where in actual tx execution the state keys provided by action 1 will be made available to action 2.

Reply:
Can leave this for now.. if we see issues with the discrepancy in the future we can update it. May also need to add the state keys from state manager in the future.